core/src/main/java/io/questdb/cutlass/text/SerialCsvFileImporter.java

```diff
@@ -0,0 +1,147 @@
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cutlass.text;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.CairoSecurityContext;
+import io.questdb.cairo.PartitionBy;
+import io.questdb.cairo.security.AllowAllCairoSecurityContext;
+import io.questdb.cairo.sql.ExecutionCircuitBreaker;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import io.questdb.std.Os;
+import io.questdb.std.Unsafe;
+import io.questdb.std.str.Path;
+
+import java.io.Closeable;
+
+public final class SerialCsvFileImporter implements Closeable {
+    private final int sqlCopyBufferSize;
+    private final CharSequence sqlCopyInputRoot;
+    private final FilesFacade ff;
+    private Path path;
+    private final CairoSecurityContext securityContext;
+    private TextLoader textLoader;
+    private CharSequence tableName;
+    private CharSequence fileName;
+    private boolean isHeaderFlag;
+    private ParallelCsvFileImporter.PhaseStatusReporter statusReporter;
+    private ExecutionCircuitBreaker circuitBreaker;
+    private int partitionBy;
+    private CharSequence timestampColumn;
+    private int atomicity;
+
+    public SerialCsvFileImporter(CairoEngine engine) {
+        CairoConfiguration configuration = engine.getConfiguration();
+        this.sqlCopyBufferSize = configuration.getSqlCopyBufferSize();
+        this.sqlCopyInputRoot = configuration.getSqlCopyInputRoot();
+        this.path = new Path();
+        this.ff = configuration.getFilesFacade();
+        this.textLoader = new TextLoader(engine);
+        this.securityContext = AllowAllCairoSecurityContext.INSTANCE;
+    }
+
+    @Override
+    public void close() {
+        path = Misc.free(path);
+        textLoader = Misc.free(textLoader);
+    }
+
+    public void of(String tableName, String fileName, int partitionBy, CharSequence timestampColumn, boolean isHeaderFlag, int atomicity, ExecutionCircuitBreaker circuitBreaker) {
+        this.tableName = tableName;
+        this.fileName = fileName;
+        this.partitionBy = partitionBy;
```

Comment:
Do we need partitionBy field here? It would always be NONE, isn't it?

Reply:
good one!